packages/back-end/src/models/FactMetricModel.ts

```diff
@@ -26,6 +32,55 @@ const BaseClass = MakeModelClass({
   readonlyFields: ["datasource"],
 });
 
+// extra checks on user filter
+function validateUserFilter({
+  metricType,
+  numerator,
+  factTable,
+}: {
+  metricType: FactMetricType;
+  numerator: ColumnRef;
+  factTable: FactTableInterface;
+}): void {
+  // error if one is specified but not the other
+  if (!!numerator.aggregateFilter !== !!numerator.aggregateFilterColumn) {
```

Comment:
Does this break the UI if you have an aggregate filter and then remove it? Does the column value stick around and cause this check to break?

Reply:
Yes, it does break the UI and it also breaks it if you add one then switch metric types.

I'll sanitize the fields on save.